test/unit/detail/version_check_test.hpp

```diff
@@ -167,36 +158,32 @@ struct version_check : public ::testing::Test
     }
 };
 
-struct sanity_checks : public version_check
-{};
-
 //------------------------------------------------------------------------------
 // sanity checks
 //------------------------------------------------------------------------------
 
 // even if the homedir might not be writable at least the tmp dir should be
-TEST_F(sanity_checks, path_availability)
+TEST_F(version_check_test, sanity_path_availability)
 {
     EXPECT_FALSE(app_tmp_path().empty()) << "No writable directory found. All other tests cannot be trusted!";
 }
 
-TEST_F(sanity_checks, create_and_delete_files)
+TEST_F(version_check_test, sanity_create_and_delete_files)
 {
     EXPECT_TRUE(create_file(app_version_filename(), "20.5.9"));
-    EXPECT_TRUE(create_file(app_timestamp_filename(), current_unix_timestamp()));
-
     EXPECT_TRUE(std::filesystem::exists(app_version_filename()));
+
+    EXPECT_TRUE(create_file(app_timestamp_filename(), current_unix_timestamp()));
     EXPECT_TRUE(std::filesystem::exists(app_timestamp_filename()));
 
     EXPECT_TRUE(remove_files_from_path()); // clear files again
     EXPECT_FALSE(std::filesystem::exists(app_version_filename()));
     EXPECT_FALSE(std::filesystem::exists(app_timestamp_filename()));
 }
 
-TEST_F(sanity_checks, cookie)
+TEST_F(version_check_test, sanity_cookie)
 {
-    char const * argv[3] = {app_name.c_str(), OPTION_VERSION_CHECK, OPTION_ON};
-    auto [out, err, app_call_succeeded] = simulate_parser(3, argv);
+    auto [out, err, app_call_succeeded] = simulate_parser(OPTION_VERSION_CHECK, OPTION_ON, "-f");
```

Comment:
former test did not have the additional `"-f"` argument ?

Reply:
Yep, and then it would just call the short help and exit :D